rust/otap-dataflow/crates/otap/src/otap_grpc/otlp/server.rs

```diff
@@ -34,10 +34,12 @@ use tonic::server::{Grpc, NamedService, UnaryService};
 /// generally optional depending on wait_for_result: true, we do not
 /// create or use the state when ack/nack is not required.
 #[derive(Clone)]
-pub struct SharedState(Arc<Mutex<SlotsState<oneshot::Sender<Result<(), NackMsg<OtapPdata>>>>>>);
+pub struct SharedState(
```

Comment:
In this PR, I made them visible to OTAP Receiver so that it can use them. These structs could be moved to a shared location as they are used by both OTLP and OTAP receiver. We can do that in another PR. I wanted to keep the diff smaller for this PR.